### PR TITLE
Nightly Fuzz CI

### PR DESCRIPTION
**Pull request by vyzaldysanchez**

### CODE REVIEW

File: .github/workflows/nightlyfuzz.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        with:
+          path: "${{ steps.get-corpus-dir.outputs.corpus_dir }}"
+          # We need to ensure uniqueness of the key, as saving to a key more than once will fail (see Save corpus step).
+          # We never expect a cache hit with the key but we do expect a hit with the restore-keys prefix that is going
+          # to match the latest cache that has that prefix.
+          key: "nightlyfuzz-corpus-${{ github.run_id }}-${{ github.run_attempt }}"
+          restore-keys: "nightlyfuzz-corpus-"
+      - name: "Run native fuzzers"
+        # Fuzz for 5 hours, after 6 hours the action would time out
+        run: "cd fuzz && ./fuzz_all_native.py --seconds 18000"
```
Comment (jmank88) [inline .github/workflows/nightlyfuzz.yml:42]: Might be too long :shrug:

**Comment by jmank88** [comment]
LGTM!
Is there a plan to make this reusable for other repos, so we don't have to copy it around?

File: .github/workflows/nightlyfuzz.yml (hunk 1 above)
Comment (vyzaldysanchez) [reply .github/workflows/nightlyfuzz.yml:42]: What time would you suggest us to do?

**Comment by vyzaldysanchez** [comment]
@jmank88 I'll be working with RelEng to make a reusable action.

File: .github/workflows/nightlyfuzz.yml (hunk 1 above)
Comment (jmank88) [reply .github/workflows/nightlyfuzz.yml:42]: Maybe 30m or 1h to start?

Comment (vyzaldysanchez) [reply .github/workflows/nightlyfuzz.yml:42]: I think that's reasonable!

**Review by jmank88 — APPROVED** [review]

**Pull request merged**